### PR TITLE
[Chore] update to cue v0.4.0

### PR DESCRIPTION
**Pull request by ying-jeanne**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/cmd/grafana-server/commands/cli.go [hunk 1]
```diff
@@ -44,22 +44,27 @@ func (e exitWithCode) Error() string {
...
+		profile     = serverFs.Bool("profile", false, "Turn on pprof profiling")
+		profileAddr = serverFs.String("profile-addr", "localhost", "Define custom address for profiling")
+		profilePort = serverFs.Uint64("profile-port", 6060, "Define custom port for profiling")
+		tracing     = serverFs.Bool("tracing", false, "Turn on tracing")
+		tracingFile = serverFs.String("tracing-file", "trace.out", "Define tracing output file")
 	)
-
-	flag.Parse()
+	switch os.Args[1] {
+	case "server":
```
Comment (ying-jeanne) [inline pkg/cmd/grafana-server/commands/cli.go:63]: here we actually don't need it if we don't want any sub-command in the future, just leave it to make the flag set usage more standard. I don't have strong opinion towards it, totally fine with a simple serverFs.Parse(os.Args[1:])

Comment (marefr) [reply pkg/cmd/grafana-server/commands/cli.go:63]: Skip it for now. Taking height for something potentially happening in the future feels bad.

**Review by marefr — CHANGES_REQUESTED** [review]
Need to test/verify that this still works as expected: https://github.com/grafana/grafana/blob/0e60abfe00511d554ee28dafc8a324684dc7cc46/pkg/cmd/grafana-server/commands/cli.go#L164

Noticed there was some linting issues as well.

**Comment by ying-jeanne** [comment]
> Need to test/verify that this still works as expected:
> 
> https://github.com/grafana/grafana/blob/0e60abfe00511d554ee28dafc8a324684dc7cc46/pkg/cmd/grafana-server/commands/cli.go#L164
> 
> Noticed there was some linting issues as well.

might because i downgraded my go version for validating the private release, fixed.

**Comment by sdboyer** [comment]
Seems good so far, takes care of the global flag issue.

When it comes to actually upgrading CUE, i think it's sufficient to only change our use of `cue.Runtime` to `cue.Context`, as recommended), and any changes that directly entails. (Like, i'm not sure how much we rely on `cue.Instance`, but there's no need to get rid of all references to it in this PR, even though it's now deprecated)

**Review by marefr — APPROVED** [review]
LGTM

**Pull request merged**